### PR TITLE
Fancy environment

### PR DESCRIPTION
**Pull request by idreessamadi**
here the XML file for Our Emulator's environment.

Screenshot (22)

Screenshot (23)

Screenshot (21)

this one is a bit cocky if you don't like it I will remove it.
Screenshot (24)

### CODE REVIEW

**Comment by platisd** [comment]
lol nice :+1:

File: share/smce/Torch/Torch/Data/UI/smartCarEnvironment.xml [hunk 1]
```diff
@@ -0,0 +1,3612 @@
...
+	<attribute name="Snap Threshold" value="5" />
+	<attribute name="Elapsed Time" value="5133.34" />
+	<attribute name="Next Replicated Node ID" value="216" />
+	<attribute name="Next Replicated Component ID" value="593" />
+	<attribute name="Next Local Node ID" value="16777370" />
+	<attribute name="Next Local Component ID" value="16777494" />
+	<attribute name="Variables" />
+	<attribute name="Variable Names" value="" />
+	<component type="Octree" id="1" />
+	<component type="DebugRenderer" id="2" />
```
Comment (RuthgerD) [inline share/smce/Torch/Torch/Data/UI/smartCarEnvironment.xml:15]: Urho doesnt document this well but don't think we need DebugRenderer

File: src/app/UrhoApp.cxx [hunk 2]
```diff
@@ -66,18 +66,10 @@ void UrhoApp::Stop() {}
...
-    plane_node->SetScale(Urho3D::Vector3(200, 0, 200));
-    auto* const planeObject = plane_node->CreateComponent<Urho3D::StaticModel>();
-    planeObject->SetModel(cache->GetResource<Urho3D::Model>("Models/Cone.mdl"));
-
-    Urho3D::Node* lightNode = m_scene->CreateChild("Sunlight");
-    auto* const light = lightNode->CreateComponent<Urho3D::Light>();
-    light->SetLightType(Urho3D::LIGHT_DIRECTIONAL);
+    Urho3D::ResourceCache* plane_node = GetSubsystem<Urho3D::ResourceCache>();
+    m_scene = new Urho3D::Scene(context_);
+    Urho3D::SharedPtr<Urho3D::File> file = cache->GetFile("Torch/Data/UI/smartCarEnvironment.xml");
```
Comment (RuthgerD) [inline src/app/UrhoApp.cxx:71]: You deleted important components like TorchMenu and EmulGlue, TorchMenu adds to cache path for Torch/ without it urho woudnt be able to find the xml file (just tested it)

File: src/app/UrhoApp.cxx [hunk 3]
```diff
@@ -66,18 +66,10 @@ void UrhoApp::Stop() {}
...
-    Urho3D::Node* plane_node = m_scene->CreateChild("Ground");
-    plane_node->SetScale(Urho3D::Vector3(200, 0, 200));
-    auto* const planeObject = plane_node->CreateComponent<Urho3D::StaticModel>();
-    planeObject->SetModel(cache->GetResource<Urho3D::Model>("Models/Cone.mdl"));
-
-    Urho3D::Node* lightNode = m_scene->CreateChild("Sunlight");
-    auto* const light = lightNode->CreateComponent<Urho3D::Light>();
-    light->SetLightType(Urho3D::LIGHT_DIRECTIONAL);
+    Urho3D::ResourceCache* plane_node = GetSubsystem<Urho3D::ResourceCache>();
+    m_scene = new Urho3D::Scene(context_);
```
Comment (RuthgerD) [inline src/app/UrhoApp.cxx:70]: `m_scene = Urho3D::MakeShared<Urho3D::Scene>(context_);`

**Comment by RuthgerD** [comment]
Would also be good to move the xml to its own folder since its not UI

File: src/app/UrhoApp.cxx (hunk 2 above)
Comment (idreessamadi) [reply src/app/UrhoApp.cxx:71]: that was a mistake I will add them back

**Comment by RuthgerD** [comment]
For tracability: scene xml loading currently has a problem where it disables all events for the gui

**Comment by RuthgerD** [comment]
Would like to have the floor texture be a little more high resolution, and the text is quite blurry with some visual glitches: either remove it or make it look good :)

**Comment by RuthgerD** [comment]
image
Floating mushrooms

**Comment by RuthgerD** [comment]
forgot to remove ` CMakeSettings.json`.. my bad for not noticing ;?

**Pull request merged**